### PR TITLE
Add RewrapManyDataKeyProseTest

### PR DESCRIPTION
**Pull request by katcharov**
JAVA-4698

### CODE REVIEW

File: driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.Map;
+import java.util.Set;
+
+import static com.mongodb.ClusterFixture.hasEncryptionTestsEnabled;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.Fixture.getMongoClient;
+import static com.mongodb.client.Fixture.getMongoClientSettingsBuilder;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public abstract class AbstractRewrapManyDataKeyProseTest {
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java:47]: Rename to `AbstractClientEncryptionRewrapManyDataKeyProseTest` to aid future discovery

File: driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.mongodb.ClusterFixture.hasEncryptionTestsEnabled;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.Fixture.getMongoClient;
+import static com.mongodb.client.Fixture.getMongoClientSettingsBuilder;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java:46]: Lets add a comment to the prose test in the spec

File: driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/RewrapManyDataKeyProseTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.mongodb.ClientEncryptionSettings;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.AbstractRewrapManyDataKeyProseTest;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.vault.ClientEncryption;
+import com.mongodb.client.vault.ClientEncryptions;
+import com.mongodb.reactivestreams.client.syncadapter.SyncMongoClient;
+
+public class RewrapManyDataKeyProseTest extends AbstractRewrapManyDataKeyProseTest {
```
Comment (rozza) [inline driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/RewrapManyDataKeyProseTest.java:27]: Rename to `ClientEncryptionRewrapManyDataKeyProseTest` to aid future discovery

**Review by rozza — CHANGES_REQUESTED** [review]
Couple of minor naming changes, to follow current conventions.

Other than that LGTM

Note: I've added JAVA-4879 to track improving the discoverability of prose tests and to define a clearer naming convention / grouping. As I'm not sure all prose tests have been implemented.

File: driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/RewrapManyDataKeyProseTest.java (hunk 3 above)
Comment (katcharov) [reply driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/RewrapManyDataKeyProseTest.java]: Many tests are named `AbstractClientSideEncryption...` - is this just a minor inconsistency, or is there some difference?

File: driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java (hunk 1 above)
Comment (katcharov) [reply driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java]: Many tests are named `AbstractClientSideEncryption...` - is this just a minor inconsistency, or is there some difference?

Comment (rozza) [reply driver-sync/src/test/functional/com/mongodb/client/AbstractRewrapManyDataKeyProseTest.java]: This is the weirdness that I hope JAVA-4879 will clean up. 

Probably a mix of inconsistent naming and grouping for the CSFLE Prose tests - its not clear which ones map to which prose tests.

**Pull request merged**